crawler/crawlutils.py

```diff
@@ -539,19 +569,5 @@ def snapshot(
         if frequency < 0 or should_exit:
             logger.info('Bye')
             break
-        elif frequency == 0:
-            continue
-
-        if next_iteration_time is None:
-            next_iteration_time = snapshot_time + frequency
-        else:
-            next_iteration_time = next_iteration_time + frequency
-
-        while next_iteration_time + frequency < time.time():
-            next_iteration_time = next_iteration_time + frequency
-
-        time_to_sleep = next_iteration_time - time.time()
-        if time_to_sleep > 0:
-            time.sleep(time_to_sleep)
```

Comment:
for event design, there's no sleep. There's only an event timeout. 
I had put it in the code comment, it is even more trickier with events.
eg. it took a snapshot at 't1' for 2 containers 'c1' and 'c2'.
at 't1+1' another container 'c3' is created, in this case, it will break out of the wait and snapshot all three 'c1, c2, c3' containers at t1+1. 

i have put it in the TODO, to handle such timing cases.